examples/tensorflow/language-modeling-tpu/prepare_tfrecord_shards.py

```diff
@@ -180,7 +180,7 @@ def group_texts(examples):
                 out_file.write(example)
             print("Wrote file {} containing {} records".format(filename, records_containing))
 
-        shard_count += 1
+        shard_count += 1  # noqa: SIM113
```

Comment:
The need for `noqa` here makes me very nervous! This kind of pattern is not uncommon, and users may be very confused if Ruff autofixes their code or requires `noqa` comments for it.

The other Ruff SIM fixes are okay I think, but cc @arthurzucker @cyrilvallez since this modifies the library code style

Reply:
`enumerate` is generally desired. Let me try to fix this example.

Reply:
Fine, let me remove SIM113.